src/main/java/io/zksync/signer/EthSigner.java

```diff
@@ -1,120 +1,102 @@
-    }
-
-    public EthSignature signMessage(String message) {
-        return signMessage(message, true);
-    }
-
-    public EthSignature signMessage(String message, boolean addPrefix) {
-        System.out.println("Eth message: " + message);
-
-        Sign.SignatureData sig = addPrefix ?
-            Sign.signPrefixedMessage(message.getBytes(), credentials.getEcKeyPair()) :
-            Sign.signMessage(message.getBytes(), credentials.getEcKeyPair());
-
-        final ByteArrayOutputStream output = new ByteArrayOutputStream();
-
-        try {
-            output.write(sig.getR());
-            output.write(sig.getS());
-            output.write(sig.getV());
-        } catch (IOException e) {
-            throw new ZkSyncException("Error when creating ETH signature", e);
-        }
-
-        final String signature = Numeric.toHexString(output.toByteArray());
-
-        return EthSignature.builder().signature(signature).type(EthSignature.SignatureType.EthereumSignature).build();
-    }
-
-    public boolean verifySignature(EthSignature signature, String message) throws SignatureException {
-        return verifySignature(signature, message, true);
-    }
-
-    public boolean verifySignature(EthSignature signature, String message, boolean prefixed) throws SignatureException {
-        byte[] sig = Numeric.hexStringToByteArray(signature.getSignature());
-        Sign.SignatureData signatureData = new Sign.SignatureData(
-            Arrays.copyOfRange(sig, 64, 65),
-            Arrays.copyOfRange(sig, 0, 32),
-            Arrays.copyOfRange(sig, 32, 64)
-        );
-        BigInteger publicKey = prefixed ?
-            Sign.signedPrefixedMessageToKey(message.getBytes(), signatureData) :
-            Sign.signedMessageToKey(message.getBytes(), signatureData);
-        return credentials.getEcKeyPair().getPublicKey().equals(publicKey);
-    }
-
-    private static Credentials generateCredentialsFromMnemonic(String mnemonic, int accountIndex) {
-        //m/44'/60'/0'/0 derivation path
-        int[] derivationPath = {44 | Bip32ECKeyPair.HARDENED_BIT, 60 | Bip32ECKeyPair.HARDENED_BIT, 0 | Bip32ECKeyPair.HARDENED_BIT, 0,accountIndex};
+import org.web3j.crypto.Credentials;
 
-        // Generate a BIP32 master keypair from the mnemonic phrase
-        Bip32ECKeyPair masterKeypair = Bip32ECKeyPair.generateKeyPair(
-                MnemonicUtils.generateSeed(mnemonic, ""));
+import io.zksync.domain.token.Token;
 
-        // Derive the keypair using the derivation path
-        Bip32ECKeyPair  derivedKeyPair = Bip32ECKeyPair.deriveKeyPair(masterKeypair, derivationPath);
+public interface EthSigner {
+    
+    /**
+     * Get wallet address
+     * 
+     * @return Address in hex string
+     */
+    String getAddress();
+
+    /**
+     * Get internal credentials. @see org.web3j.crypto.Credentials
+     * 
+     * @return Credentials object that contains private and public key pair
+     */
+    Credentials getCredentials();
```

Comment:
The interface is created mainly for the purpose of enabling JSON-RPC signers.

These are remote signers, and they can not return their credentials.